eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script automates the patch release process for a given library.
+# Note that this assumes that the only changes needed in the patch are upgrades to the dependencies.
+# It makes all the necessary changes and pushes them in an upstream branch which can then be used to trigger the release
+# Please follow the necessary steps for the release - validating all the changes, trigger the release pipeline, approve the release etc.
+# It takes in the following arguments
+# The script takes the following arguments
+# 1. ArtifactName - Name of the library that needs to be patched. This is a required parameter.
+# 2. ServiceDirectoryName - The directory name under 'sdk' folder where the source code of the library is present. This is a required parameter.
+# 3. ReleaseVersion - The latest release version of the library which is to be patched. This is a required parameter.
+# 4. PatchVersion - The version of the patch. This is a required parameter.
+#                   TODO: We can remove the need for this parameter by intepreting this from the first version.
+# 5. BranchName - The name of the remote branch where the changes will be pushed. We push the changes to the 'upstream remote'. This is not a required parameter.
+#                 In case the argument is not provided the branch name is release/{ArtifactName}.
+#
+# Example:  .\eng\scripts\Generate-Patch.ps1 -ArtifactName azure-mixedreality-remoterendering -ServiceDirectory remoterendering -ReleaseVersion 1.0.0 -PatchVersion 1.0.1
+# This creates a remote branch "release/azure-mixedreality-remoterendering" with all the necessary changes.
+
+param(
+  [Parameter(Mandatory=$true)][string]$ArtifactName,
+  [Parameter(Mandatory=$true)][string]$ServiceDirectoryName,
+  [Parameter(Mandatory=$true)][string]$ReleaseVersion,
+  [Parameter(Mandatory=$true)][string]$PatchVersion,
+  [Parameter(Mandatory=$false)][string]$BranchName
+)
+
+function TestPathThrow($Path) {
+  if(!(Test-Path $Path)) {
+   LogError "$Path not found"
+   exit
+  }
+}
+
+Write-Information "PS Script Root is: $PSScriptRoot"
+Write-Information "ArtifactName is: $ArtifactName"
+Write-Information "ReleaseVersion is: $ReleaseVersion"
+Write-Information "ServiceDirectoryName is: $ServiceDirectoryName"
+Write-Information "PatchVersion is: $PatchVersion"
+
+if($null -eq $BranchName) {
+  $BranchName = "release/$artifactName.ToLower()"
+}
+Write-Information "BranchName is: $branchName"
+
+$RepoRoot = Resolve-Path "${PSScriptRoot}..\..\.."
+$EngDir = Join-Path $RepoRoot "eng"
+$EngCommonScriptsDir = Join-Path $EngDir "common" "scripts"
+$SdkDirPath = Join-Path $RepoRoot "sdk"
+$ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
+$ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
```

Comment:
Do we have checks to ensure none of the artifact directory in our repo have upper-cases? The script should allow for case maintainence where ever possible. I chose toLower() for branchnames in case the user does not provide it.